classes/Tools.php

```diff
@@ -517,7 +517,7 @@ public static function getValue($key, $default_value = false)
         }
 
         if (is_string($value)) {
-            return stripslashes(urldecode(preg_replace('/((\%5C0+)|(\%00+))/i', '', urlencode($value))));
+            return urldecode(preg_replace('/((\%5C0+)|(\%00+))/i', '', urlencode($value)));
```

Comment:
It looks like this code replaces `"\0"` and `"\\0"`. Maybe it could be optimized using `str_replace()`?

```suggestion
            return str_replace(["\\0", "\0"], '', $value);
```

Reply:
Agreed, just didn't want to touch something here, and still don't understand why they use urlencode :/

Reply:
Maybe it has something to do with multibyte encoding?

Reply:
I update the code because "\0" is enough.

Reply:
ping @eternoendless

Reply:
You never changed it @PierreRambaud :(